python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -1,30 +1,135 @@
+from ._interop_helpers cimport (
+    _release_schema,
+    _release_array,
+    _metadata_to_libcudf,
+)
+from ._interop_helpers import ColumnMetadata
 
-from functools import cache
+import functools
 
 
 __all__ = ["Column", "ListColumnView", "is_c_contiguous"]
 
 
+class _ArrowLikeMeta(type):
+    def __subclasscheck__(cls, other):
+        return hasattr(other, "__arrow_c_array__")
+
+
+class _ArrowLike(metaclass=_ArrowLikeMeta):
+    pass
+
+
+cdef class _ArrowColumnHolder:
+    """A holder for an Arrow column for gpumemoryview lifetime management."""
+    cdef unique_ptr[arrow_column] col
+
+
+cdef class OwnerWithCAI:
+    """An interface for column view's data with gpumemoryview via CAI."""
+    @staticmethod
+    cdef create(column_view cv, object owner):
+        obj = OwnerWithCAI()
+        obj.owner = owner
+        cdef int size
+        cdef column_view offsets_column
+        cdef unique_ptr[scalar] last_offset
+        if cv.type().id() == type_id.EMPTY:
+            size = cv.size()
+        elif is_fixed_width(cv.type()) or is_fixed_point(cv.type()):
+            size = cv.size() * cpp_size_of(cv.type())
+        elif cv.type().id() == type_id.STRING:
+            # The size of the character array in the parent is the offsets size
+            num_children = cv.num_children()
+            size = 0
+            # A strings column with no children is created for empty/all null
+            if num_children:
+                offsets_column = cv.child(0)
+                last_offset = get_element(offsets_column, offsets_column.size() - 1)
+                size = (<numeric_scalar[size_type] *> last_offset.get()).value()
+        else:
+            # All other types store data in the children, so the parent size is 0
+            size = 0
+
+        obj.cai = {
+            "shape": (size,),
+            "strides": None,
+            # For the purposes in this function, just treat all of the types as byte
+            # streams of the appropriate size. This matches what we currently get from
+            # rmm.DeviceBuffer
+            "typestr": "|u1",
+            "data": (<Py_ssize_t> cv.head[char](), False),
+            "version": 3,
+        }
+        return obj
+
+    @property
+    def __cuda_array_interface__(self):
+        return self.cai
+
+
+cdef class OwnerMaskWithCAI:
+    """An interface for column view's null mask with gpumemoryview via CAI."""
+    @staticmethod
+    cdef create(column_view cv, object owner):
+        obj = OwnerMaskWithCAI()
+        obj.owner = owner
+
+        obj.cai = {
+            "shape": (bitmask_allocation_size_bytes(cv.size()),),
+            "strides": None,
+            # For the purposes in this function, just treat all of the types as byte
+            # streams of the appropriate size. This matches what we currently get from
+            # rmm.DeviceBuffer
+            "typestr": "|u1",
+            "data": (<Py_ssize_t> cv.null_mask(), False),
```

Comment:
nitpick: Does `uintptr_t` work here?
```suggestion
            "data": (<uintptr_t> cv.null_mask(), False),
```

Reply:
Sorry I know we mentioned this before

Reply:
Yes, it does work. Done in https://github.com/rapidsai/cudf/pull/18402/commits/932b0ad4035088ccfd8a1300c6fa140ad42ee13d